src/lib/message/L1Transaction.ts

```diff
@@ -200,8 +240,36 @@ export class L1TransactionReceipt implements TransactionReceipt {
     const provider = SignerProviderUtils.getProviderOrThrow(l2SignerOrProvider)
     const network = await getL2Network(provider)
     const chainID = network.chainID.toString()
-    const events = this.getMessageEvents()
 
+    // Classic events
+    if (this.blockNumber < network.nitroGenesisBlock) {
+      const { messageNum } = this.getInboxMessageDeliveredEvents()[0]
+
+      return [
+        new ClassicL1ToL2MessageReader(
+          provider,
+          BigNumber.from(chainID).toNumber(),
+          '0x',
+          messageNum,
+          BigNumber.from(0),
+          // No message params required for classic events
+          // Populate with empty data
+          {
```

Comment:
I think you could remove this from the constructor args, and just set up an empty messageData in the constructor itself. You can do the same for other empty args, like the base fee and sender.